M2/Macaulay2/m2/webapp.m2

```diff
@@ -3,28 +3,37 @@
 -- htmlWithTex Thing produces some valid html code with possible TeX code
 -- topLevelMode=WebApp produces that plus possible pure text coming from the system
 -- hence, requires tags to help the browser app distinguish html from text
+webAppTags := apply((17,18,19,20,28,29,30,31,17),ascii);
     (webAppEndTag,            -- closing tag ~ </span>
 	webAppHtmlTag,        -- indicates what follows is HTML ~ <span class='M2Html'>
 	webAppOutputTag,      -- it's html but it's output ~ <span class='M2Html M2Output'>
 	webAppInputTag,       -- it's text but it's input ~ <span class='M2Input'>
 	webAppInputContdTag,  -- text, continuation of input
+	webAppUrlTag,         -- used internally
 	webAppTextTag,        -- other text ~ <span class='M2Text'>
 	webAppTexTag,         -- TeX start ~ \(
 	webAppTexEndTag       -- TeX end ~ \)
-	)=apply((17,18,19,20,28,30,31,17),ascii);
+	)=webAppTags;
 
 texMathStart := "\\(";
 texMathEnd := "\\)";
-texWrap := x -> texMathStart | texMath x | texMathEnd; -- similar to 'tex', but only used by webapp.m2 to avoid thread-safety issues
+texWrap := x -> (  -- similar to 'tex', but only used by webapp.m2 to avoid thread-safety issues -- TODO: rewrite in a thread-safe way
+    y := texMath x;
+    if class y =!= String then error "invalid texMath output";
+    texMathStart | y | texMathEnd
+    )
```

Comment:
What aspect of this is not thread safe?

Reply:
We discussed this during my previous PR (can't find the conversation for some reason); you're the one who convinced me non thread-safe code was bad. I'm merely writing a note to myself to eventually fix this; no actual change here.